docs/guide/execution-modes.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Job execution modes
+---
+
+# Choosing an execution mode for your job
+
+## Understanding execution modes
+
+Jobq allows you to run your jobs in different execution modes (through the `jobq submit --mode` parameter), in some cases even without any changes to your code.
+
+Execution modes determine how your job is executed, where it runs, and how resources are managed. The choice of execution mode depends on your requirements, such as the scale of your job, the resources it needs, and the frameworks and libraries you are using in your code.
+
+For example, if you are running a small job that does not require any external dependencies, you can submit your job as a "plain" Kubernetes batch job. However, if you are performing large-scale machine learning training, you might want to use a Ray job to distribute your workload across multiple compute nodes.
+
+### Local execution
+
+The most basic execution mode is `local`, which runs your job as a Python function on the same machine where the jobq client is executed. This mode does not provide any of the benefits such as automatic resource management, but it is the fastest way to get started.
+
+As an additional benefit, since you are running your job locally, you can easily debug your code and test it on a small scale before scaling it up to a cluster-based execution mode.
+Breakpoints, logging, and other debugging tools work as expected in this mode.
+
+### Cluster-based execution
+
+The more advanced execution modes require a backend to run your job and a Kubernetes cluster where the job can be scheduled and executed. All Kubernetes-based execution modes are built on top of [Kueue](https://kueue.sigs.k8s.io/), a Kubernetes-native job scheduling system, and extend its underlying functionality.
+
+Currently, jobq supports two cluster-based execution modes:
+
+-   **[Kueue Mode](#kubernetes-batch-jobs)**: This mode runs your job as a Kubernetes built-in `batch/v1/Job`.
```

Comment:
Maybe it's worth mentioning that Kueue mode requires a Kueue installation + an existing cluster queue (if it's not supposed to be Inadmissible)?

Reply:
Valid point, but I think that's sufficiently covered by the installation guide already, which includes Kueue as its prerequisites.

I would like not to duplicate too much content in the how-to guides to keep them focused on the topic at hand.